File: python_projects/readability.py
def main():
    text = input("Text: ")
    let_num = letters(text)
    sen_num = sentences(text)
    wor_num = words(text)
    l = (let_num / wor_num) * 100
    s = (sen_num / wor_num) * 100
    index = 0.0588 * l - 0.296 * s - 15.8
    final = round(index)
    if final < 1:
        print("Before Grade 1")
    
    elif final >= 16:
        print("Grade 16+")

    else:
        print(f"Grade {final}")        


def letters(text):
    count = 0
    for letter in text:
        if letter.isalpha() == True:
            count += 1
    return count

def words(text):
    count = 0
    for letter in text:
        if letter == " ":
            count +=1
    return count + 1        

def sentences(text):
    count = 0
    for letter in text:
        if letter in [".","?","!"]:
            count +=1
    return count

main()

File: C_projects/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>
#include "../samples.h"
int letters(char *text);
int words(char *text);
int sentences(char *text);

int main(void)
{
    char *x = get_string("Text: ");
    int num_l = letters(x);
    int num_w = words(x);
    int num_s = sentences(x);
    double l = (num_l / (double)num_w)*100;//average letters per 100 words
    double s = (num_s / (double)num_w)*100;//average sentences per 100 words
    double index = 0.0588 * l - 0.296 * s - 15.8;
    int final = round(index);//for round to nearest integer
    
    if(final < 1)
    {
        printf("Before Grade 1");
    }
    else if(final >= 16)
    {
        printf("Grade 16+");
    }
    else
    {
        printf("Grade %i",final);
    }
    free(x);
          
}

int letters(char *text)
{
    int letters = 0;
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if((text[i] >= 'a'&& text[i] <= 'z')||(text[i] >= 'A'&& text[i] <= 'Z'))
        {
            letters += 1;
        }
        
    }
    return letters;
}

int words(char *text)
{
    int words = 0;
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if(text[i] == ' ')
        {
            words += 1;
        }
    }
    return words+1;
    
}

int sentences(char *text)
{
    int sentences = 0;
    for(int i = 0, n = strlen(text); i < n; i++)
    {
        if((text[i] == '.')||(text[i] == '!')||(text[i] == '?'))
        {
            sentences += 1;
        }
    }
    return sentences;
    
}

